internal/usecase/client/pendo/pendo_client_test.go

```diff
@@ -0,0 +1,374 @@
+	cfg := helperPendoConfig()
+	client := helperNewPendo(cfg, func(req *http.Request) *http.Response {
+		return nil
+	})
+
+	// Error path 1
+	ctx := app_context.CtxWithLog(context.TODO(), slog.Default())
+	metrics := make(pendo.SetMetadataRequest, 0, 1)
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "test-visitor-id",
+		Values: map[string]any{
+			"test-field-1": 2,
+			"test-field-2": "anotherString",
+		},
+	})
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "thisVisitorDoesNotExist",
+	})
+	resp, err := client.SetMetadata(ctx, kind, group, metrics)
+	require.EqualError(t, err, "Post \"http://localhost:8031/pendo/v1/metadata/account/custom/value\": http: RoundTripper implementation (pendo.RoundTripFunc) returned a nil *Response with a nil error")
+	require.Nil(t, resp)
+}
+
+func TestSetMetadataErrorHttp(t *testing.T) {
+	kind := pendo.KindAccount
+	group := pendo.Group("custom")
+	// https://hassansin.github.io/Unit-Testing-http-client-in-Go#2-by-replacing-httptransport
+	cfg := helperPendoConfig()
+	client := helperNewPendo(cfg, func(req *http.Request) *http.Response {
+		return &http.Response{
+			StatusCode: http.StatusBadGateway,
+			Body:       http.NoBody,
+			Header:     make(http.Header),
+		}
+	})
+
+	// Error path 1
+	ctx := app_context.CtxWithLog(context.TODO(), slog.Default())
+	metrics := make(pendo.SetMetadataRequest, 0, 1)
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "test-visitor-id",
+		Values: map[string]any{
+			"test-field-1": 2,
+			"test-field-2": "anotherString",
+		},
+	})
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "thisVisitorDoesNotExist",
+	})
+	resp, err := client.SetMetadata(ctx, kind, group, metrics)
+	require.EqualError(t, err, "Pendo SetMetadata request unsuccessful: status: 502")
+	require.Nil(t, resp)
+}
+
+func TestSetMetadataErrorUnmarshalling(t *testing.T) {
+	kind := pendo.KindAccount
+	group := pendo.Group("custom")
+	// https://hassansin.github.io/Unit-Testing-http-client-in-Go#2-by-replacing-httptransport
+	cfg := helperPendoConfig()
+	client := helperNewPendo(cfg, func(req *http.Request) *http.Response {
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(bytes.NewBufferString("{")),
+			Header:     make(http.Header),
+		}
+	})
+
+	// Error path 1
+	ctx := app_context.CtxWithLog(context.TODO(), slog.Default())
+	metrics := make(pendo.SetMetadataRequest, 0, 1)
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "test-visitor-id",
+		Values: map[string]any{
+			"test-field-1": 2,
+			"test-field-2": "anotherString",
+		},
+	})
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "thisVisitorDoesNotExist",
+	})
+	resp, err := client.SetMetadata(ctx, kind, group, metrics)
+	require.EqualError(t, err, "unexpected end of JSON input")
```

Comment:
Following up the comment with higher and lower-level logging.  With that logic, here, I'd expect error: "parsing response for SetMetadata"